nameserver/entry_test.go

```diff
@@ -0,0 +1,113 @@
+package nameserver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/weaveworks/weave/net/address"
+	"github.com/weaveworks/weave/router"
+)
+
+func TestAdd(t *testing.T) {
+	oldNow := now
+	defer func() { now = oldNow }()
+	now = func() int64 { return 1234 }
```

Comment:
I've done a similar thing here: https://github.com/weaveworks/scope/blob/master/probe/docker/registry_test.go#L118 but its pretty ugly.  Unfortunately code repetition is common in Go.